source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -897,7 +923,7 @@ void ESolver_KS_LCAO<TK, TR>::iter_finish(int iter)
     ModuleBase::TITLE("ESolver_KS_LCAO", "iter_finish");
 
     // mix density matrix
-    if (GlobalV::MIXING_RESTART > 0 && iter >= this->p_chgmix->mixing_restart && GlobalV::MIXING_DMR )
+    if (GlobalV::MIXING_RESTART > 0 && this->p_chgmix->mixing_restart_count > 0 && GlobalV::MIXING_DMR )
```

Comment:
in fact, I would suggest these kinds of details disappear in esolver and be moved away to somewhere else

Reply:
Solved.

Reply:
I decide to keep this verbosity in esolver to remind others `mixing_dmr` is only called in this case.